src/node_crypto.cc

```diff
@@ -741,6 +743,63 @@ void SecureContext::AddCRL(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void UseExtraCaCerts(const char* file) {
+  free(extra_root_certs_file);
+  extra_root_certs_file = file ? strdup(file) : nullptr;
+}
+
+static void EmitWarning(Environment* env, const char* fmt, ...) {
+  char warning[1024];
+  va_list ap;
+
+  va_start(ap, fmt);
+  vsnprintf(warning, sizeof(warning), fmt, ap);
+  va_end(ap);
+
+  HandleScope handle_scope(env->isolate());
+  Context::Scope context_scope(env->context());
+
+  Local<Object> process = env->process_object();
+  Local<Value> f =
+      process->Get(FIXED_ONE_BYTE_STRING(env->isolate(), "emitWarning"));
+  Local<Value> arg = node::OneByteString(env->isolate(), warning);
+
+  // MakeCallback() unneeded, because emitWarning is internal code, it calls
+  // process.emit('warning', ..), but does so on the nextTick.
+  if (!f->IsFunction()) return;
+  f.As<v8::Function>()->Call(process, 1, &arg);
+}
+
+static void LoadExtraCaCerts(Environment* env) {
+  if (!extra_root_certs_file)
+    return;
+
+  ERR_clear_error();
+  MarkPopErrorOnReturn mark_pop_error_on_return;
```

Comment:
@bnoordhuis I don't know if this is worth leaving in, the only caller of this function uses `ClearErrorOnReturn`, so this pop isn't necessary. Still, it should be fast, and extra certs are only loaded once, anyhow.